Source/NSTableView.m

```diff
@@ -6940,6 +6948,88 @@ - (NSView *) makeViewWithIdentifier: (NSUserInterfaceItemIdentifier)identifier o
   return view;
 }
 
+- (id) _prototypeCellViewFromTableColumn: (NSTableColumn *)tb
+{
+  NSArray *protoCellViews = [tb _prototypeCellViews];
+  id view = nil;
+  
+  // it seems there is always one prototype...
+  if ([protoCellViews count] > 0)
+    {
+      view = [protoCellViews objectAtIndex: 0];
+      view = [view copy]; // instantiate the prototype...
+    }
+
+  return view;
+}
+
+- (NSTableRowView *) rowViewAtRow: (NSInteger)row makeIfNecessary: (BOOL)flag
+{
+  NSTableRowView *rv = nil;
+
+  if (_viewBased == YES)
+    {
+      // NSLog(@"_rowViews = %@", _rowViews);
+      if (row < [_rowViews count])
+	{
+	  rv = [_rowViews objectAtIndex: row];
+	}
+      
+      if (rv == nil)
+	{
+	  if (flag == YES)
+	    {
+	      if ([_delegate respondsToSelector: @selector(tableView:rowViewForRow:)])
+		{
+		  rv = [_delegate tableView: self rowViewForRow: row];
+		}
+	    }
+	  
+	  if (rv == nil)
```

Comment:
This block belongs into the `if (flag == YES)` condition.

Reply:
True, done.